src/generators/AbstractDependencyManager.ts

```diff
@@ -0,0 +1,16 @@
+export abstract class AbstractDependencyManager {
+  constructor(
+    public dependencies: string[] = []
+  ) {}
```

Comment:
Each dependency manager should have access to the generator options, by this you can avoid passing option to the `TypeScriptDependencyManager` methods.

Reply:
I tried to avoid it for the complexity of generics and because providing the options on creation makes it hard to define a default value in the same manner we currently use. 

Do you think it's necessary?

Reply:
Okay added the options as I saw it was an annoying experience for the user to keep track of it when adding dependencies 😄